File: AnalizatorMreznogPrometa/src/Loader/TvzCSVLoader.java
package Loader;

import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.LinkedList;

public class TvzCSVLoader implements ILoader {
	private int n;
	private String fileName;
	public int KolicinaTestnihPodataka = 0;

	public TvzCSVLoader(String fileName, int n) {
		this.fileName = System.getProperty("user.dir") + fileName;
		this.n = n;
	}

	@Override
	public LinkedList<PrometData> Load(String[] onlyLoad) {
		LinkedList<PrometData> promet = new LinkedList<PrometData>();
		int validanPromet = 0;
		int validanOdgovor = 0;
		int falsePromet = 0;
		try (BufferedReader br = new BufferedReader(new FileReader(fileName))) {
			String line;

			while ((line = br.readLine()) != null) {
				validanPromet++;
				if (validanPromet >= n) {
					break;
				}
				// System.out.println(line);
				String[] parsedLine = line.split(";");
				PrometData linijaPrometa = new PrometData();
				// System.out.print(parsedLine[0]);
				// System.out.print(parsedLine[1]);
				// System.out.print(parsedLine[2]);
				// System.out.print(parsedLine[3]);
				// System.out.print(parsedLine[4]);
				// System.out.print(parsedLine[5]);
				// System.out.print(parsedLine[6]);
				// System.out.print("\n");
				boolean loadOk = false;
				for (String vrsteLoad : onlyLoad) {
					if (vrsteLoad.equals(parsedLine[8])) {
						loadOk = true;
					}
				}

				if (loadOk) {
					linijaPrometa.Time = parsedLine[3];
					linijaPrometa.Vrsta = parsedLine[8];
					linijaPrometa.SEQ = parsedLine[9];
					linijaPrometa.ASEQ = Long.parseLong(parsedLine[10]);
					linijaPrometa.SourceIP = parsedLine[12];
					linijaPrometa.SourcePort = parsedLine[13];
					linijaPrometa.DestinationIP = parsedLine[14];
					linijaPrometa.DestinationPort = parsedLine[15];
					//KolicinaTestnihPodataka je za procjenu false positive, a to
					//se gleda broj gresaka/broj ACK(*) zahtjeva
					if(!linijaPrometa.Vrsta.equals("SYN-ACK")){
						KolicinaTestnihPodataka++;
					}
					promet.addLast(linijaPrometa);
										}
			}
		} catch (IOException e) {
			e.printStackTrace();
		}
		
		System.out.println("KolicinaTestnihPodataka=" + KolicinaTestnihPodataka);
		return promet;
	}

	@Override
	public int GetKolicinaTestnihPodataka() {
		return KolicinaTestnihPodataka;
	}

}

File: Maybe-master/DataLoader/TvzCSVLoader.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DataLoader
{
    public class TvzCSVLoader : ILoader
    {
        private int n;
        private string fileName;
        public int KolicinaTestnogPrometa { get; set; }
        public TvzCSVLoader(string fileName, int n)
        {
            string path = Directory.GetParent(Environment.CurrentDirectory).Parent.Parent.FullName;
            this.fileName = path + fileName;
            this.n = n;
        }

        public LinkedList<PrometData> Load(string[] olnyLoad)
        {
            LinkedList<PrometData> promet = new LinkedList<PrometData>();
            int validanPromet = 0;

            using (StreamReader sr = new StreamReader(fileName))
            {
                int brojac = 0;
                while (!sr.EndOfStream)
                {
                    if (brojac >= n)
                    {
                        break;
                    }
                    brojac++;
                    string[] line = sr.ReadLine().Split(';');

                    if (olnyLoad.Contains(line[8]))
                    {
                        PrometData linija = new PrometData()
                        {
                            Time = line[3],
                            Vrsta = line[8],
                            SEQ = line[9],
                            ASEQ = Double.Parse(line[10]),
                            DestinationIP = line[14],
                            DestinationPort = line[15],
                            SourcePort = line[13],
                            SourceIP = line[12]
                        };
                        promet.AddLast(linija);
                        if (!linija.Vrsta.Equals("SYN-ACK"))
                        {
                              validanPromet++;
                        }
                    }
                }
            }
            KolicinaTestnogPrometa = validanPromet;
            return promet;
        }
    }
}
